pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -738,6 +639,46 @@ func (c *Controller) getProxyServiceInstancesByEndpoint(endpoints v1.Endpoints,
 	return out
 }
 
+func (c *Controller) getProxyServiceInstancesByEndpointSlice(slice discoveryv1alpha1.EndpointSlice, proxy *model.Proxy) []*model.ServiceInstance {
+	out := make([]*model.ServiceInstance, 0)
+
+	hostname := kube.ServiceHostname(slice.Labels[discoveryv1alpha1.LabelServiceName], slice.Namespace, c.domainSuffix)
+	c.RLock()
```

Comment:
nit: you can replace the lock, get, unlock with `c.GetService(hostname)`.

Reply:
I had to change the function to return true/false depending on if its found or not, but I think thats better anyways

Reply:
Nevermind, I reverted this change. This is common between all registries (like consul) and its deprecated, so probably do not want to use/

Reply:
Not sure I understand why you needed to change the method signature ... doesn't `svc == nil` mean that it wasn't found?

Reply:
You can store nil I think, although maybe in practice that doesn't happen

Reply:
that would seem odd ... probably worth a quick scan of the code to see if we ever do this.